### PR TITLE
Make subdivided hyper cube/rectangle work for general Triangulation.

### PR DESCRIPTION
**Pull request by luca-heltai**

### CODE REVIEW

File: include/deal.II/grid/grid_generator.h [hunk 1]
```diff
@@ -89,8 +89,8 @@ namespace GridGenerator
    *
    * @note The triangulation needs to be void upon calling this function.
    */
-  template <int dim>
-  void subdivided_hyper_cube (Triangulation<dim>  &tria,
+  template <int dim, int spacedim>
+  void subdivided_hyper_cube (Triangulation<dim,spacedim>  &tria,
                               const unsigned int   repetitions,
                               const double         left = 0.,
                               const double         right= 1.);
```
Comment (bangerth) [inline include/deal.II/grid/grid_generator.h]: Please document how `left` and `right` are interpreted if `dim<spacedim`. Are only the first `dim` components set to be between left and right? All components?

File: doc/news/changes.h [hunk 2]
```diff
@@ -428,6 +428,13 @@ inconvenience this causes.
 <h3>Specific improvements</h3>
 
 <ol>
+  <li> New: The GridGenerator::subdivided_hyper_cube() and 
+  GridGenerator::subdivided_hyper_rectangle() now work also for codimension
+  one and two Triangulation();
```
Comment (bangerth) [inline doc/news/changes.h]: No need for parentheses after the class name.

**Comment by bangerth** [comment]
Please make the two changes and merge yourself. Thanks!

File: include/deal.II/grid/grid_generator.h (hunk 1 above)
Comment (luca-heltai) [inline include/deal.II/grid/grid_generator.h]: That's already in the documentation, which was copied from hyper_cube, but was only made to work with one template argument: 

~~~

   *
   * If spacedim=dim+1 the same mesh as in the case spacedim=dim is created,
   * but the vertices have an additional coordinate =0. So, if dim=1 one
   * obtains line along the x axis in the xy plane, and if dim=3 one obtains a
   * square in lying in the xy plane in 3d space.
   *
~~~

Comment (bangerth) [inline include/deal.II/grid/grid_generator.h]: Ah, ok. Thanks! (Then just one issue to address :-)

**Comment by tamiko** [comment]
> Please make the two changes and merge yourself. Thanks!

* and maybe add a trivial testcase that outputs a codimension one and
  two mesh? :-]

**Pull request merged**